plugins/autoscroll.koplugin/main.lua

```diff
@@ -0,0 +1,136 @@
+local Device = require("device")
+local Event = require("ui/event")
+local PluginShare = require("pluginshare")
+local UIManager = require("ui/uimanager")
+local WidgetContainer = require("ui/widget/container/widgetcontainer")
+local logger = require("logger")
+local _ = require("gettext")
+local T = require("ffi/util").template
+
+local AutoScroll = WidgetContainer:new{
+    name = 'autoscroll',
+    is_doc_only = true,
+    auto_scroll_sec = G_reader_settings:readSetting("auto_scroll_timeout_seconds") or 0,
+    enabled = G_reader_settings:isTrue("auto_scroll_enabled"),
+    settings_id = 0,
+    last_action_sec = os.time(),
+}
+
+function AutoScroll:_enabled()
+    return self.enabled and self.auto_scroll_sec > 0
+end
+
+function AutoScroll:_schedule(settings_id)
+    if not self:_enabled() then
+        logger.dbg("AutoScroll:_schedule is disabled")
+        return
+    end
+    if self.settings_id ~= settings_id then
+        logger.dbg("AutoScroll:_schedule registered settings_id ",
+                   settings_id,
+                   " does not equal to current one ",
+                   self.settings_id)
+        return
+    end
+
+    local delay
+
+    if PluginShare.pause_auto_scroll then
```

Comment:
A bit unclear to me: are you using that evil BackgroundRunner? Or doing the UIManager.schedule / unschedule yourself?
(looks like the later - but that PluginShare made me thought initially you went with BackgroundRunner).
Anyway, is that `PluginShare.pause_auto_scroll` needed ? A left-over from copied plugin? Something for the future to stop it from elsewhere?

Reply:
Oh yeah, a vestigial function. This is a KeepAlive hook for the AutoSuspend plugin.

Reply:
Come to think of it, it should still be kept. This one and AutoSuspend reset with an InputEvent hook, but this plugin doesn't emit input events, only `GotoViewRel`.